shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1272,65 +1284,185 @@ - (void)updateViewportPadding {
 
 #pragma mark - Keyboard events
 
+- (void)keyboardWillShowNotification:(NSNotification*)notification {
+  // Immediately prior to a docked keyboard being shown or when a keyboard goes from
+  // undocked/floating to docked, this notification is triggered.
+  [self handleKeyboardNotification:notification];
+}
+
 - (void)keyboardWillChangeFrame:(NSNotification*)notification {
-  NSDictionary* info = [notification userInfo];
+  // Immediately prior to a change in keyboard frame, this notification is triggered.
+  [self handleKeyboardNotification:notification];
+}
 
-  // Ignore keyboard notifications related to other apps.
-  id isLocal = info[UIKeyboardIsLocalUserInfoKey];
-  if (isLocal && ![isLocal boolValue]) {
+- (void)keyboardWillBeHidden:(NSNotification*)notification {
+  // When keyboard is hidden or undocked, this notification will be triggered.
```

Comment:
are there examples when this is not triggered (from your design doc)

Reply:
One example of this not being triggered is when the keyboard is switch from docked to floating. I'll add that to the design doc.

Reply:
can you add comment here too? this part can get tricky.

Reply:
done